src/sentry/static/sentry/app/components/modals/inviteMembersModal/index.tsx

```diff
@@ -0,0 +1,391 @@
+      sendingInvites: false,
+    });
+
+  sendInvite = async (invite: NoramlizedInvite) => {
+    const {slug} = this.props.organization;
+    const data = {
+      email: invite.email,
+      teams: [...invite.teams],
+      role: invite.role,
+    };
+
+    this.setState(s => ({
+      inviteStatus: {...s.inviteStatus, [invite.email]: {sent: false}},
+    }));
+
+    try {
+      await this.api.requestPromise(`/organizations/${slug}/members/`, {
+        method: 'POST',
+        data,
+      });
+    } catch (err) {
+      const errorResponse = err.responseJSON;
+
+      // Use the email error message if available. This inconsistently is
+      // returned as either a list of errors for the field, or a single error.
+      const emailError =
+        !errorResponse || !errorResponse.email
+          ? false
+          : Array.isArray(errorResponse.email)
+          ? errorResponse.email[0]
+          : errorResponse.email;
+
+      const error = emailError || t('Could not invite user');
+
+      this.setState(s => ({
+        inviteStatus: {...s.inviteStatus, [invite.email]: {sent: false, error}},
+      }));
+      return;
+    }
+
+    this.setState(s => ({
+      inviteStatus: {...s.inviteStatus, [invite.email]: {sent: true}},
+    }));
+  };
+
+  sendInvites = async () => {
+    this.setState({sendingInvites: true});
+    await Promise.all(this.invites.map(this.sendInvite));
+    this.setState({sendingInvites: false, complete: true});
+  };
+
+  addInviteRow = () =>
+    this.setState(s => ({pendingInvites: [...s.pendingInvites, this.inviteTemplate]}));
+
+  setEmails(emails: string[], index: number) {
+    this.setState(s => {
+      const pendingInvites = [...s.pendingInvites];
+      pendingInvites[index] = {...pendingInvites[index], emails: new Set(emails)};
+
+      return {pendingInvites};
+    });
+  }
+
+  setTeams(teams: string[], index: number) {
+    this.setState(s => {
+      const pendingInvites = [...s.pendingInvites];
+      pendingInvites[index] = {...pendingInvites[index], teams: new Set(teams)};
+
+      return {pendingInvites};
+    });
+  }
+
+  setRole(role: string, index: number) {
+    this.setState(s => {
+      const pendingInvites = [...s.pendingInvites];
+      pendingInvites[index] = {...pendingInvites[index], role};
+
+      return {pendingInvites};
+    });
+  }
+
+  removeInviteRow(index: number) {
+    this.setState(s => {
+      const pendingInvites = [...s.pendingInvites];
+      pendingInvites.splice(index, 1);
+      return {pendingInvites};
+    });
+  }
+
+  get invites(): NoramlizedInvite[] {
+    return this.state.pendingInvites.reduce<NoramlizedInvite[]>(
+      (acc, row) => [
+        ...acc,
+        ...[...row.emails].map(email => ({email, teams: row.teams, role: row.role})),
+      ],
+      []
+    );
+  }
+
+  get hasDuplicateEmails() {
+    const emails = this.invites.map(inv => inv.email);
+    return emails.length !== new Set(emails).size;
+  }
+
+  get isValid() {
```

Comment:
When I read this, I wasn't immediately sure what this referred to

```suggestion
  get isValidInvites() {
```